presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -1251,6 +1258,11 @@ private Partitioning createPartitioning(List<Symbol> partitioningColumns)
             PartitioningHandle partitioningHandle = metadata.getPartitioningHandleForExchange(session, partitioningProviderCatalog, hashPartitionCount, partitioningTypes);
             return Partitioning.create(partitioningHandle, partitioningColumns);
         }
+
+        private Scope getExchangeScope()
```

Comment:
it sounds weird for deciding whether to materialize or not to be called getScope().  maybe `getExchangeType()`?

Reply:
That's a problem. The ExchangeNode already has something else named "type". The type is whether the exchange is REPLICATE, PARTITION, GATHER